pennylane/templates/subroutines/qrom.py

```diff
@@ -306,3 +327,154 @@ def wires(self):
     def clean(self):
         """Boolean to select the version of QROM."""
         return self.hyperparameters["clean"]
+
+
+def _qrom_decomposition_resources(
+    num_bitstrings, num_control_wires, num_target_wires, num_work_wires, clean
+):
+    if num_control_wires == 0:
```

Comment:
Similar to the comment I made in your other PR, we generally don't want to replicate the decomposition logic in the resource function unless there's absolutely no other way to do this. Maybe you can take a look at
https://github.com/PennyLaneAI/pennylane/blob/5c7e16bc85df482eaa718dbcb9e972e021f27005/pennylane/labs/resource_estimation/templates/subroutines.py#L850
and see if it fits here.